src/qililab/qprogram/qblox_compiler.py

```diff
@@ -247,33 +255,35 @@ def _handle_parallel(self, element: Parallel):
     def _handle_average(self, element: Average):
         for bus in self._buses:
             qpy_loop = QPyProgram.Loop(name=f"avg_{self._buses[bus].average_counter}", begin=element.shots)
-            qpy_loop.append_component(
-                component=QPyInstructions.WaitSync(wait_time=QbloxCompiler.minimum_wait_duration),
-                bot_position=len(qpy_loop.components),
-            )
             self._buses[bus].qpy_block_stack[-1].append_component(qpy_loop)
             self._buses[bus].qpy_block_stack.append(qpy_loop)
             self._buses[bus].average_counter += 1
         return True
 
+    def _handle_infinite_loop(self, _: InfiniteLoop):
+        for bus in self._buses:
```

Comment:
what is the purpose of infinite loops?

Reply:
What do you mean by purpose? 😅

Reply:
I assume it's so we can do something like qblox's infinite loops
```
seq_prog = """
      wait_sync 4

loop: play    0,0,1200
      jmp     @loop
"""
```
but if it's like that, is killing the process the only way to stop it?

Reply:
Yes for Qblox, this is the only way. Either manually or through some timeout mechanism. Quantum Machines support infinite loops natively though.